projects/hathor-wallet-headless/0003-fireblocks-integration.md

```diff
@@ -0,0 +1,130 @@
+# Summary
+[summary]: #summary
+
+Hathor headless wallet integration with Fireblocks using RAW Signing API.
+
+# Motivation
+[motivation]: #motivation
+
+Fireblocks manages the private key so removing the private key from the headless instance will allow users to run the headless in a more secure manner.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+<details>
+
+ <summary><code>POST</code> <code><b>/wallet/fireblocks/start</b></code> <code>(Start a wallet with the fireblocks integration)</code></summary>
+
+This API will use the wallet's `setExternalTxSigningMethod` to register a method that will start a client for fireblocks and use Fireblock's RAW signing API to sign the transaction.
+
+##### Parameters
+
+> | name | type | data type | description | location |
+> | --- | --- | --- | --- | --- |
+> | xpub-id | required | string | The id of the xpub in the config | body | 
+> | wallet-id | required | string | create a wallet with this id | body |
+> | raw | required | boolean | use RAW signing | body |
+
+##### Responses
+
+> | http code | content-type | response |
+> | --- | --- | --- |
+> | `200` | `application/json` | `{"success":true}` |
+> | `400` | `application/json` | `{"success": false, "message":"Bad Request"}` |
+
+##### Example cURL
+
+> ```javascript
+>  curl -X POST -H "Content-Type: application/json" --data '{"xpub-id": "cafe", "wallet-id": "cafe", "raw": true}' 'http://localhost:8000/fireblocks/start'
+> ```
+
+</details>
+
+Fireblocks has documented a few possibilities for integrations, we will use the [REST API](https://developers.fireblocks.com/docs/rest-api-guide) to implement our client, this means that we need to implement the authorization workflow and API calls.
+
+## Fireblocks Client
+
+### BIP44 path derivation
+
+Fireblocks follows a derivation path very similar to BIP44 but without any hardened derivations.
+This is why we can use the root xPub from the console to generate the account level xPub, even though it is not a BIP44 account level it will work exactly the same with our wallet-lib.
+
+The usual account derivation path of `m/44'/280'/0'` will be replaced with `m/44/280/0`, we will need to create a script that derives the root xPub to the fireblocks account level path.

```

Comment:
Fireblocks does not release the private key, having the root xPub derive to all coins means that if the user leaks his root xPub he will lose privacy, meaning anyone can know his balance on any coin.

We only require the "account adjacent" path xPub, if leaked it can only reveal the Hathor wallet balance.

Reply:
I see. I believe adding this to the design text would be enough to make this point clear, then.

That way it's clear from the start how the security will be kept for future readers of the RFCs

Reply:
Done